variables.tf

```diff
@@ -95,6 +95,56 @@ variable "managed_identities" {
   description = "Managed identities to be created for the resource."
 }
 
+variable "node_pools" {
+  type = map(object({
+    vm_size             = string
+    enable_auto_scaling = bool
+    max_count           = number
+    min_count           = number
+    node_count          = number
+    os_sku              = string
+    mode                = optional(string, "User")
+    os_disk_size_gb     = optional(number, null)
+    tags                = optional(map(string), {})
+    zones               = optional(string)
+  }))
+  default = {
+    "1" = {
```

Comment:
To avoid confusion you could call it "workload1" "workload2" and so on

Reply:
added this to the when the resource is being created